src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerTest.java

```diff
@@ -350,4 +335,51 @@ void GIVEN_log_manager_in_errored_state_WHEN_restarted_THEN_logs_upload_is_reatt
         }
         assertEquals(1, allFiles.size());
     }
-}
+
+    @Test
+    void GIVEN_files_randomly_named_WHEN_logs_are_uploaded_to_cloud_THEN_active_file_exists_and_all_data_uploaded()
+            throws Exception {
+        when(cloudWatchLogsClient.putLogEvents(any(PutLogEventsRequest.class))).thenReturn(
+                PutLogEventsResponse.builder().nextSequenceToken("nextToken").build());
+
+        tempDirectoryPath = Files.createDirectory(tempRootDir.resolve("logs"));
+        LogConfig.getRootLogConfig().setLevel(Level.TRACE);
+        LogConfig.getRootLogConfig().setStore(LogStore.FILE);
+        LogManager.getLogConfigurations().putIfAbsent("UserComponentB",
+                newLogConfigFromRootConfig(LogConfigUpdate.builder().fileName("UserComponentB.log").build()));
+
+        for (int i = 0; i < 6; i++) {
+            String randomName1 = UUID.randomUUID().toString();
```

Comment:
I don't just want the files to be randomly named and then uploaded. I want that, and then I want the files to be randomly re-named and verify that we _do not_ upload them again.